src/confirm.rs

```diff
@@ -0,0 +1,116 @@
+use std::time::Duration;
+use futures::{Future, Stream};
+use api::{Eth, EthFilter, Namespace};
+use types::{H256, U256, TransactionRequest};
+use {Transport, Error};
+
+pub fn wait_for_confirmations<'a, T, F, V>(transport: T, poll_interval: Duration, confirmations: usize, check: V)
```

Comment:
I think it would be better if it was part of some kind of namespace.
The idea is that you're creating `Web3` object and you can do everything from there.
Now you would need to create two separate transports (one for Web3 and one for `wait_for_confirmations`) and use the function separately.

Maybe put it directly to `Web3Main`?